openedx_events/learning/data.py

```diff
@@ -331,6 +331,24 @@ class ExamAttemptData:
     requesting_user = attr.ib(type=UserData, default=None)
 
 
+@attr.s(frozen=True)
+class CourseAccessRoleData:
```

Comment:
Reading the `ManageStudentsPermissionData` class defined below makes me wonder if we should refrain from using the word "role" in this class name. I know you've talked to more people about the roles + permissions work though, so I trust your judgement!

Reply:
thanks. Yup the conclusion we came to was to define this as edx-platform exists today rather than trying to guess the future. I kept the name specific to the LMS model so it's use is clear instead of a ambiguous half-measure that might be reusable in the future.

There's no longer a near term path to the permission changes getting merged as we had originally wanted to account for. I'm not sure manage students will even end up being the correct permission when that the time does come to use permissions over roles.

Reply:
Got it, appreciate the explanation! This looks good then 👍